library/kani_macros/src/sysroot/contracts/shared.rs

```diff
@@ -144,3 +147,171 @@ pub fn try_as_result_assign(stmt: &syn::Stmt) -> Option<&syn::LocalInit> {
 pub fn try_as_result_assign_mut(stmt: &mut syn::Stmt) -> Option<&mut syn::LocalInit> {
     try_as_result_assign_pat!(stmt, as_mut)
 }
+
+/// This function goes through the vector of statements and counts the number of variables
+/// with idents prefixed by `remember_kani_internal_`. This avoid variable name collisions by
```

Comment:
Are these remember variables ever exposed to the user? If not I would prefer to use a short hash of the entire function body as a prefix or suffix rather than a count.

Reply:
@JustusAdam Why would that be preferable in this case?

Reply:
Counting is brittle, plus code that matches over syntax (e.g. this function) are effort to maintain if the AST format changes. We should already have a routine that calculated a unique hash (it may even be stored in the generators already). Using a hash would completely eliminate this syntax matching function.

Reply:
Now using syn::Expr's Hash trait to hash the contents of `old(expr)`. If there is a hash collision, it would be because the exprs are identical, which means we can use the same variable for both expressions.